### ISSUE TITLE
[Bug]: Caused by java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by ajaybirla-fabzen** [OneSignal/OneSignal-Unity-SDK#681]
### What happened?

The application crashes whenever it receives a notification.

### Steps to reproduce?

```Markdown
1. Launch the application.
2. Send a notification to the application.
3. Click on the notification.
4. Observe the application crash.
```

### What did you expect to happen?

I expect the app to not crash whenever it receives a notification.

### Unity version

2021.3.17f1

### OneSignal Unity SDK version

5.0.5

### Platform

Android

### Relevant log output

```Shell
Caused by java.util.ConcurrentModificationException:
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at com.onesignal.common.events.EventProducer.fire(EventProducer.kt:43)
       at com.onesignal.session.internal.session.impl.SessionService.backgroundRun(SessionService.kt:66)
       at com.onesignal.core.internal.background.impl.BackgroundManager$runBackgroundServices$2$1.invokeSuspend(BackgroundManager.kt:124)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:367)
       at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
       at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
       at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
       at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
       at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
       at kotlinx.coroutines.BuildersKt.launch(:1)
       at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
       at kotlinx.coroutines.BuildersKt.launch$default(:1)
       at com.onesignal.core.internal.background.impl.BackgroundManager$runBackgroundServices$2.invokeSuspend(BackgroundManager.kt:120)
       at com.onesignal.core.internal.background.impl.BackgroundManager$runBackgroundServices$2.invoke(:8)
       at com.onesignal.core.internal.background.impl.BackgroundManager$runBackgroundServices$2.invoke(:4)
       at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
       at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
       at com.onesignal.core.internal.background.impl.BackgroundManager.runBackgroundServices(BackgroundManager.kt:117)
       at com.onesignal.core.services.SyncJobService$onStartJob$1.invokeSuspend(SyncJobService.kt:48)
       at com.onesignal.core.services.SyncJobService$onStartJob$1.invoke(:8)
       at com.onesignal.core.services.SyncJobService$onStartJob$1.invoke(:2)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1$1.invokeSuspend(ThreadUtils.kt:61)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jinliu9508** [comment]
@ajaybirla-fabzen Thank you for providing the trace stack! This is a known issue to us and we are expecting to solve this in the next release.

**Comment by nan-li** [comment]
The fix has been released in [5.0.6 Release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.0.6). Please update and reach out if you are still seeing any issues. Thanks!
